### PR TITLE
Api 20613 rswag documentation per env

### PR DESCRIPTION
**Pull request by FonzMP**
## Description of change

Introduces new rake task.
ex: `rake rswag:claims_api:build`

## Things to know about this PR

**Things I like:** 
- Runs for dev and production
- We'll serve the appropriate swagger.json per environment (dev/staging -> v2/dev/swagger and sandbox/prod ->  v2/production/swagger)

**Things I don't like:**
- Runs v1 twice.
- The old rake task still brings in 5103 cause it's not `document: false`, but don't really have a fix for it.
- Not real keen on the `TextHelpers.new` in all the files. <- any ideas?
- Would prefer this to only run v2, but with the `rswag_config.rb` set up, it's a bit tougher to implement without burning some more time.

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/docs/v2/api_controller.rb [hunk 1]
```diff
@@ -5,7 +5,9 @@ module Docs
     module V2
       class ApiController < ClaimsApi::Docs::ApiController
         def index
-          swagger = JSON.parse(File.read(ClaimsApi::Engine.root.join('app/swagger/claims_api/v2/swagger.json')))
+          env = request.base_url == ('https://api.va.gov' || 'https://sandbox-api.va.gov') ? '' : 'dev/'
+
+          swagger = JSON.parse(File.read(ClaimsApi::Engine.root.join("app/swagger/claims_api/v2/#{env}swagger.json")))
```
Comment (stiehlrod) [inline modules/claims_api/app/controllers/claims_api/docs/v2/api_controller.rb:10]: {env}/swagger.json")))

File: spec/rswag_override.rb [hunk 2]
```diff
@@ -3,6 +3,34 @@
...
+      def example_group_finished(notification)
+        metadata = if RSPEC_VERSION > 2
+          notification.group.metadata
+        else
+          notification.metadata
+        end
+
+        # !metadata[:document] won't work, since nil means we should generate
+        # docs.
+        return if metadata[ENV['DOCUMENTATION_ENVIRONMENT']&.to_sym] == false
```
Comment (FonzMP) [inline spec/rswag_override.rb:15]: this is the only modification from the original rswag file.

File: rakelib/rswag.rake [hunk 3]
```diff
@@ -25,6 +25,19 @@ namespace :rswag do
...
+    desc 'Generate rswag docs by environment for the claims_api'
+    task build: :environment do
+      ENV['PATTERN'] = 'modules/claims_api/spec/requests/**/*_spec.rb'
+      ENV['RAILS_MODULE'] = 'claims_api'
+      ENV['SWAGGER_DRY_RUN'] = '0'
+      %w[dev production].each do |environment|
+        ENV['DOCUMENTATION_ENVIRONMENT'] = environment
+        Rake::Task['rswag:specs:swaggerize'].invoke
+        %w[v1 v2].each { |version| strip_swagger_base_path(version, (version.eql?('v2') ? environment : nil)) }
+        Rake::Task['rswag:specs:swaggerize'].reenable
```
Comment (FonzMP) [inline rakelib/rswag.rake:38]: have to reenable to run 2x. This was a joy to figure out.

**Comment by stiehlrod** [comment]
Should we add to the README?

**Review by stiehlrod — APPROVED** [review]
LGTM

**Pull request merged**